SU2_CFD/src/variables/CNEMONSVariable.cpp

```diff
@@ -102,6 +104,14 @@ bool CNEMONSVariable::SetPrimVar(unsigned long iPoint, CFluidModel *FluidModel)
   for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
     DiffusionCoeff(iPoint, iSpecies) = Ds[iSpecies];
 
+  su2double T   =  Primitive(iPoint,nSpecies);
+  su2double Tve =  Primitive(iPoint,nSpecies+1);
+
+  su2double* val_eves = GetEve(iPoint);
+  hs = fluidmodel->ComputeSpeciesEnthalpy(T, Tve, val_eves);
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
```

Comment:
Try changing the local "iVars" and "iSpecies" declared in the NEMO variable member functions to unsigned long to see if we can fix some of those LGTM warnings. (e.g. line 82 of this file)

Reply:
Sure will do. Out of curiosity, why does this throw a warning? Is it because if the iVar is unsigned short, there are values that can be stored in an unsigned long that it cannot hold?

Reply:
That's the only reason I can think of, it is general good practice to avoid mixing types.